src/prommis/nanofiltration/diafiltration_flowsheet_two_salt.py

```diff
@@ -74,11 +75,62 @@ def main():
+            # skip check at x=0 as the concentration is expected to be 0 and the
+            # diffusion coefficient calculation is not needed
+            if x == 0:
+                pass
+            elif not (50 < value(m.fs.membrane.membrane_conc_mol_lithium[x, z]) < 80):
+                raise ValueError(
+                    "Membrane concentration for lithium ("
+                    f"{value(m.fs.membrane.membrane_conc_mol_lithium[x, z])} mM at "
+                    f"x={x * value(m.fs.membrane.total_module_length)} m and "
+                    f"z={z * value(m.fs.membrane.total_membrane_thickness)} m) is outside "
+                    "of the valid range for the diffusion coefficient approximations "
+                    "(50-80 mM). The linearized approximation should be re-calculated."
+                )
+    if m.fs.membrane.config.charged_membrane:
+        for x in m.fs.membrane.dimensionless_module_length:
+            for z in m.fs.membrane.dimensionless_membrane_thickness:
+                # skip check at x=0 as the concentration is expected to be 0 and the
+                # diffusion coefficient calculation is not needed
+                if x == 0:
+                    pass
+                elif not (
+                    80 < value(m.fs.membrane.membrane_conc_mol_cobalt[x, z]) < 110
+                ):
+                    raise ValueError(
+                        "Membrane concentration for cobalt ("
+                        f"{value(m.fs.membrane.membrane_conc_mol_cobalt[x, z])} mM at "
+                        f"x={x * value(m.fs.membrane.total_module_length)} m and "
+                        f"z={z * value(m.fs.membrane.total_membrane_thickness)} m) is outside "
+                        "of the valid range for the diffusion coefficient approximations "
+                        "(80-110 mM). The linearized approximation should be re-calculated."
+                    )
+    else:
+        for x in m.fs.membrane.dimensionless_module_length:
+            for z in m.fs.membrane.dimensionless_membrane_thickness:
+                # skip check at x=0 as the concentration is expected to be 0 and the
+                # diffusion coefficient calculation is not needed
+                if x == 0:
+                    pass
+                elif not (
+                    50 < value(m.fs.membrane.membrane_conc_mol_cobalt[x, z]) < 80
+                ):
+                    raise ValueError(
+                        "Membrane concentration for cobalt ("
+                        f"{value(m.fs.membrane.membrane_conc_mol_cobalt[x, z])} mM at "
+                        f"x={x * value(m.fs.membrane.total_module_length)} m and "
+                        f"z={z * value(m.fs.membrane.total_membrane_thickness)} m) is outside "
+                        "of the valid range for the diffusion coefficient approximations "
+                        "(50-80 mM). The linearized approximation should be re-calculated."
+                    )
+
```

Comment:
There's an old and little-used API for validation like this: the `model_check` method on the [unit model block](https://github.com/IDAES/idaes-pse/blob/7d19f17d8bbf3c7790ddef827751a5da3a12648c/idaes/core/base/unit_model.py#L118). It would probably be good to implement these checks by implementing a `model_check` method on your unit model.